tp3/graph_functions.py

```diff
@@ -0,0 +1,130 @@
+class FifoQueue:
+    def __init__(self):
+        self.lista = []
+
+    def get(self):
+        if not self.empty():
+            return self.lista.pop(0)
+
+    def empty(self):
+        return len(self.lista) == 0
+
+    def put(self, dato):
+        self.lista.append(dato)
+        return
+
+def make_path(parents, start, end):
+    v = start
+    parent = parents[v]
+    final_path = []
+    while v != end:
+        final_path.append((parent, v))
+        v = parent
+        parent = parents[v]
+    return final_path
+
+
+def path(graph, start, end):
+    if start not in graph or end not in graph:
+        return []
+
+    visited = {}
+    visited[end] = True
+    parents = {}
+    parents[end] = None
+    process = FifoQueue()
+    process.put(end)
+    v = None
+    while not process.empty():
+        v = process.get()
+        if v == start:
+            break
+
+        for w in graph.get_neighbours(v):
+            if w not in visited:
+                visited[w] = True
+                parents[w] = v
+                process.put(w)
+
+    if v != start:
+        return []
+
+    return make_path(parents, start, end)
+
+def actors_at_distance(graph, actor, distance):
+    """"""
+    list = n_steps(graph, actor, distance)
+    list.sort()
+    return list
+
+
+def popularity(graph, actor):
+    """"""
+    if not actor in graph:
+        return -1
+
+    actors = actors_at_distance(graph, actor, 2)
+    movies_counted = set()
+    movies_count = 0
+    for actor in actors:
+        for edge in graph.get_edges_of_vertex(actor):
+            for movie in edge.get_information():
+                if not movie in movies_counted:
+                    movies_count += 1
+                    movies_counted.add(movie)
+
+    print("total de actores", len(actors))
+    print("total de peliculas", movies_count)
```

Comment:
Mmm, no me gustan estos `prints` en la función de la librería de funciones! Y son polémicos tantos `for` anidados.